packages/plugin-vue-jsx/package.json

```diff
@@ -42,7 +42,7 @@
     "vite": "^4.4.9"
   },
   "peerDependencies": {
-    "vite": "^4.0.0",
+    "vite": "^4.0.0 || ^5.0.0-0",
```

Comment:
Should it be `5.0.0-beta.0`?

Reply:
It seems that `5.0.0-0` contains a larger range. But to be honest, I don't really know the difference between the two. 🤦‍♂️

Reply:
Never mind. Another thing that worries me is that Vite 5 is not stable yet, and we can't be sure if there will be some breaking changes.

/cc @vitejs/core

Reply:
I would think it's reasonable to have it now, to avoid duplication of Vite versions.

For the breaking changes etc, it's already indicated in Vite's version if they use `5.0.0-beta.0`, I would think if there are incompatibility issues, it should be expected when using beta releases (and we could fix later).

Reply:
I'm not sure about this.

I guess most non-trivial projects would use more than one plugin.
But we can't expect every ecosystem plugin to update its peer dependency range to a pre-release version.
Therefore, the impact of updating only the core peer dependency range would likely be very small.
On the other hand, this introduces another maintenance burden for us.

I propose fixing this issue in the userland.
If a user is curious enough to try out a beta version of Vite, I suppose they would not be frightened by a few more setup steps:
- For pnpm, there's [`pnpm.peerDependencyRules.allowedVersions`](https://pnpm.io/package_json#pnpmpeerdependencyrulesallowedversions)
- For yarn, the [`resolution` field](https://yarnpkg.com/configuration/manifest#resolutions) would most likely do the job and [`packageExtensions`](https://yarnpkg.com/configuration/yarnrc#packageExtensions) may help to fix the few edge cases.
- I'm not very familiar with npm, but [`overrides` field](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#overrides) + [`--legacy-peer-deps`](https://docs.npmjs.com/cli/v9/using-npm/config?v=true#legacy-peer-deps) are likely to solve most problems.

Reply:
Yeah I think this should also only be userland for now. In `vite-plugin-svelte`, we only bump the range for prerelease if the Vite plugin code needs to change to accommodate a breaking change, in this case we don't need yet.